### DISCUSSION TITLE
Web-based build123d editors

### DISCUSSION THREAD

**Discussion by kcleung** [gumyr/build123d#839] [Q&A]
I am going through the build123d tutorials.

At the moment, I only have access to work laptop which I do not have admin access to, so this makes it difficult for me to install a fully-functional build123d library with IDE that allows me to create, render and test build123d models.

Are there any website which hosts web-based IDEs (such as OCP CAD viewer with build123d library included) which allows me to type/upload python files, and run these files?

**Comment by lypwig** [comment]
You don't need an admin access to install build123d if you install it in a virtual environment (which is recommended in any Python project):

```py
python -m venv .venv
source .venv/bin/activate
pip install build123d ocp-vscode
```

**Comment by gumyr** [comment]
Don't most corporate security systems stop the creation of a virtual environment?

**Reply by lypwig** [reply]
I am not aware of this but maybe. @kcleung is that the case for you?
